spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/EndpointWebMvcHypermediaManagementContextConfiguration.java

```diff
@@ -139,6 +142,13 @@ public DefaultCurieProvider curieProvider(ServerProperties server,
 		return new DefaultCurieProvider("boot", new UriTemplate(path));
 	}
 
+	@Bean
+	@ConditionalOnBean(FindByIndexNameSessionRepository.class)
+	@ConditionalOnEnabledEndpoint("session")
+	public SessionMvcEndpoint sessionEndpoint() {
```

Comment:
Shouldn't this go into ```EndpointWebMvcManagementContextConfiguration```?
Making it conditional on presence of HATEOAS makes no sense to me.